packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -380,12 +371,12 @@ export const useGridRows = (
     throttledRowsChange(
       convertGridRowsPropToState({
         rows,
-        props: { rowCount: props.rowCount, getRowId: props.getRowId },
```

Comment:
I stored them in the internal cache (which was not in the state before this PR) to avoid adding them as dependencies of a `useEffect` by mistake which could cause useless tree generations.

But it is exactly the same as adding an `eslint-disable-line react-hooks/exhaustive-deps` on those `useEffect`. 

And after a quick look on this hook, it seems that we now have way less `useEffect` than before so this duplication issue should not be a big problem if we stay vigilent.

Reply:
It's rare when we need to use `eslint-disable-line react-hooks/exhaustive-deps` in the codebase. Lets see if it doesn't cause bugs keeping that way.